### ISSUE TITLE
`max-empty-lines` counts the run at the head of an embedded block as a stylesheet's and its fix treats it as the page's, so the warning is swallowed

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#585]
### What minimal example or steps are needed to reproduce the bug?

`max-empty-lines` reads the first line break of a text as closing an empty line, which is right for a stylesheet — a file opening with a break opens with an empty line — and counts the run at the head of an embedded block the same way. Its fix leaves the `raws.before` of the first node alone where the root stands in a document, since that whitespace belongs to the page around the block, so the run is reported and never written, and since a fix callback cannot decline, `--fix` reports the warning as fixed.

Measured on `177f15b` and on the branch of #481, which changes none of these rows:

| input | option | without `--fix` | with `--fix` |
| --- | --- | --- | --- |
| `<style>⏎⏎⏎a {}⏎</style>` | `1` | `3:1` | unchanged, nothing reported |
| `<style>⏎⏎a {}⏎</style>` | `0` | `2:1` | unchanged, nothing reported |
| `⏎⏎⏎a {}` — the same run at the head of a plain stylesheet | `1` | `2:1`, `3:1` | `⏎a {}` |

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": ["@stylistic/stylelint-plugin"],
	"overrides": [
		{
			"files": ["**/*.html"],
			"customSyntax": "postcss-html",
			"rules": {
				"@stylistic/max-empty-lines": 1
			}
		}
	]
}
```

### What did you expect to happen?

The run at the head of an embedded block is either counted the way the fix treats it — as the page's whitespace, one line short of a stylesheet's — or written to the count the option asks for.

### What actually happened?

The check counts it as a stylesheet's, the fix treats it as the page's, and the warning between the two is swallowed.

### Notes

Found by the review of #481. #404 is the same rule over a whitespace-only stylesheet and a different count.
